### ISSUE TITLE
[Accessibility] Focus position are reset when routing between pages

### ISSUE THREAD

**Issue by luomo-pro** [flutter/flutter#93926]
The focus of the a11y is losing state after popped from pages.

## Summary

When we're using talkback in native Android applications, jump to a new page and return, and the focus of talkback will remain the previous focus.
For example, when there are three buttons on the home page, click the second button to jump to another page. Then I press the back button to return to the previous page. The primary focus will be on the second button.

On the flutter, I found that after returning to the previous page, the focus was on the first button. It doesn't focus on the previous position, that is, the second button, like the native Android.

This should be because the flutter does not have the correct accessibility event notification, so the talkback cannot determine the position of the previous focus.

## Environment
- Flutter (Channel beta, 2.7.0-3.1.pre)
- Talkback 9.1
- Android 9.0

**Comment by luomo-pro** [comment]
Add some information.
I tested the native page Jump and the page Jump in flutter.
I found that the accessibility events triggered are different:

1. In the native, when the "WindowStateChanged" event is triggered, "ClassName" is the full package name of the jump page, and "PackageName" is the package name of the application.
This event will be triggered when the page jumps and the page returns.
However, in the "WindowStateChanged" event triggered in the flutter,Both information is missing.

2. In the flutter, the "ViewAccessibilityFocusCleared" event will be triggered twice.
This event will clear the accessibility focus.
And when the page returns, it will also be triggered twice.
In native, this event will not be triggered.

3. In the native, when the page returns, it will trigger "ViewAccessibilityFocused" to move the focus to the previous position.
In the flutter, this event is not triggered.

I suspect that the above problems may cause the flutter page to jump and fail to record the previous focus.

**Comment by luomo-pro** [comment]
I found this problem exists on both iOS and Android.
Whether it is returning to the page or closing the dialog, the accessibility focus will be on the first control of the page.
In the native iOS, use the two-finger rub gesture to return to the page, and the focus will return to the previously clicked position.
It greatly affects the user experience, such as:
1. The software displays a list of news headlines, and the user clicks on a headline to enter a new page to read, and then return.
At this point, the focus is directly on the first control of the page, instead of the news headline you just clicked. This makes it difficult for users to continue browsing from where they were just now.
2. There is a button for selecting a year. After the user clicks it, a dialog box will pop up to let him select a certain year.
When he selects a certain year in the dialog box, the text of the button for selecting the year will become the selected year.
When the dialog box is closed, the focus is on the first control of the page, which causes the user to manually browse to the select year button to determine whether he has selected the correct year.

Can this problem be handled? It will greatly improve the user experience, thank you very much!
@chunhtai

**Comment by wapscccs** [comment]
I am also a blind developer, and I also encountered the same problem when using Flutter. I hope this problem can be resolved as soon as possible, because blind users really need this

**Comment by wapscccs** [comment]
I think this is a very serious problem, can you please help 
@chunhtai

**Comment by chunhtai** [comment]
duplicates of https://github.com/flutter/flutter/issues/97522
